### ISSUE TITLE
Modal is swallowing certain events

### ISSUE THREAD

**Issue by Subaku** [kaliber5/ember-bootstrap#950]
This seems to only happen when I'm testing via `ember serve`. When I open a Modal any `@onClick` stop working but work outside of modals. I also notice that if I build and run my app (both dev and prod) outside of using `ember serve` things work normally. So maybe not an issue with ember-bootstrap?

To help reduce the problem I had copied an example from the docs that includes a close button for the modal. It exhibited the same behavior so I know it's not something funky with my app specific modals.

Note I also get no errors, warnings or deprecations via the inspector/console when doing this.

Ember: 3.14.1
Ember-Bootstrap: 3.1.0

**Comment by jelhan** [comment]
Hello @Subaku, thanks for reporting that issue. Unfortunately I didn't got yet, what you mean exactly. Could you please provide the steps to reproduce the issue? Either a CodeSandbox or repo or complete list of steps starting with `ember new <app-name>`. It should also include expected and actual results.

**Comment by Subaku** [comment]
Okay, figured it out. The cause of the issue is a weird combination of wrapping the modal in an `{{#if}}` helper and Ember-Wormhole. But first I'll clarify what my issue was for you.

When the modal opened certain components rendered would not fire their actions. I.e. buttons would do nothing. Text input elements wouldn't fire on change events to update their value etc. 

Now the issue needed two conditions to happen:
1) Let Ember-Bootstrap auto append its wormhole div which for me was to the `<body>` and outside the root element of my app.
2) Wrap the modal component in a `{{#if}}` helper tied to a boolean that you toggle on to automatically show the modal.

Looks like there's some old but still open issues regarding similar behavior in Ember-Wormhole's github:
https://github.com/yapplabs/ember-wormhole/issues/59
https://github.com/yapplabs/ember-wormhole/issues/66

I fixed my issue by manually placing the wormhole div `<div id="ember-bootstrap-wormhole"></div>` right inside my apps root element in my `application.hbs` template and removed `{{content-for "body-footer"}}` in my `index.html` which was where the div was automatically being appended. This is why when I said this only happened when serving my app via `ember serve`. I embed my app elsewhere hence the wormhole div doesn't get appended like in development.

Anyway that's what I've found. Not sure there's anything to do about it from this repo's standpoint except maybe a cautionary note perhaps. I'll leave closing or leaving this issue open up to you :)

**Comment by jelhan** [comment]
Thanks a lot for detailed explanation. Just one quick follow-up question: Do you have optional feature `default-async-observers` enabled?

**Comment by Subaku** [comment]
Negative. I did initially but that ended up causing problems as noted in #939 so I've had that option turned off since then.

**Comment by simonihmig** [comment]
@Subaku are you sure `ember-wormhole` is in use in your app? We don't use it directly, only through https://github.com/kaliber5/ember-in-element-polyfill (as we still use an older version of the polyfill). And there `ember-wormhole` is only used for versions of Ember that does not ship with `-in-element` (which was introduced in Ember 2.10!)

> I embed my app elsewhere hence the wormhole div doesn't get appended like in development

Ah, that makes sense. So if the modal is inserted outside of the application root element (which in your case is some other server-rendered div, not `body` I assume?), Ember's EventDispatcher which uses event delegation will not able to dispatch events to any components rendered outside of the application root element (where event delegation handlers are attached to).

> I fixed my issue by manually placing the wormhole div <div id="ember-bootstrap-wormhole"></div> right inside my apps root element in my application.hbs template 

Makes sense, see above. We also did this is in similar cases...

> and removed {{content-for "body-footer"}} in my index.html 

Better set `insertEmberWormholeElementToDom` to `false`, see https://www.ember-bootstrap.com/#/getting-started/setup

**tl;dr**

This is not a bug really, but rather a documentation issue, for cases where the Ember app boots in an "other" div which we do not control. Not sure where to best add instructions though 🤔

**Comment by Subaku** [comment]
> This is not a bug really, but rather a documentation issue, for cases where the Ember app boots in an "other" div which we do not control.

Agreed! Just quickly looking maybe under under the Setup section in the Getting Started portion of the docs? Could be just a few sentences bringing it to user's attention as a potential pitfall.
